llvm/lib/Target/AArch64/MCTargetDesc/AArch64InstPrinter.cpp

```diff
@@ -1348,9 +1348,14 @@ void AArch64InstPrinter::printPredicateAsCounter(const MCInst *MI,
                                                  const MCSubtargetInfo &STI,
                                                  raw_ostream &O) {
   unsigned Reg = MI->getOperand(OpNum).getReg();
+  O << "pn";
+  if (Reg >= AArch64::P0 && Reg <= AArch64::P15)
```

Comment:
Is this change needed? I would expect a predicate-as-counter to be printed as a predicate-as-mask, but not the other way around.

Reply:
Yeah you're right, It's not needed after removing the PPRPNR classes from earlier. I've simplified this to 
```c++
if (Reg < AArch64::PN0 || Reg > AArch64::PN15)
    llvm_unreachable("Unsupported predicate-as-counter register");
O << "pn" << Reg - AArch64::PN0;
```

Reply:
No it's not needed, I've removed the normal predicate logic and tidied this if block up a bit.